routing.go

```diff
@@ -445,18 +444,13 @@ func (dht *IpfsDHT) Provide(ctx context.Context, key cid.Cid, brdcst bool) (err
 		return err
 	}
 
-	mes, err := dht.makeProvRecord(keyMH)
-	if err != nil {
-		return err
-	}
-
 	wg := sync.WaitGroup{}
 	for p := range peers {
 		wg.Add(1)
 		go func(p peer.ID) {
 			defer wg.Done()
 			logger.Debugf("putProvider(%s, %s)", keyMH, p)
-			err := dht.sendMessage(ctx, p, mes)
+			err := dht.protoMessenger.PutProvider(ctx, p, keyMH, dht.host)
```

Comment:
We can if you think so. However, I put in the `host` instead of just `peer.AddrInfo` to account for:
1) You cannot currently create a valid provider record for any peer other than `host`
2) When we have signed provider records we'll be able to get them from the `host`, but not from `peer.AddrInfo`

However, given that this is a "use me if you know what you're doing" function 1 isn't a huge concern. Additionally, we likely will need to add/change the function signature anyway once we have signed provider records to accommodate rebroadcasting other people's records.

LMK what you think